lib/spack/spack/test/util/git.py

```diff
@@ -80,3 +85,21 @@ def test_pull_checkout_branch(git, tmp_path: pathlib.Path, mock_git_version_info
 
         with pytest.raises(exe.ProcessError):
             spack.util.git.pull_checkout_branch("main")
+
+
+def test_not_git_repo(tmp_path: pathlib.Path):
+    url, root = spack.util.git.git_url_root(str(tmp_path))
+    assert url is None and root is None
+
+
+def test_git_url_root(monkeypatch):
+    repo_path = os.path.dirname(os.path.abspath(__file__))
```

Comment:
this file isn't necessarily part of a git repo, better to initialize it in a temp dir.